src/BitNode/ui/BitNodeAdvancedOptions.tsx

```diff
@@ -266,12 +268,40 @@ export function BitNodeAdvancedOptions({
   callbacks,
 }: BitNodeAdvancedOptionsProps): React.ReactElement {
   const [open, setOpen] = React.useState(false);
+  const [intelligenceOverride, setIntelligenceOverride] = React.useState<string>("");
   const getSfLevel = React.useCallback(
     (sfNumber: number): number => {
       return currentSourceFiles.get(sfNumber) ?? 0;
     },
     [currentSourceFiles],
   );
+  const intelligenceOverrideTool = (
+    <>
+      <Typography component="div" display="flex" gap="1rem">
+        <Typography display="flex" alignItems="center">
+          Override Intelligence:
+        </Typography>
+        <TextField
+          sx={{ maxWidth: "4rem" }}
+          disabled={intelligenceOverride === ""}
+          value={intelligenceOverride}
+          onChange={(event) => {
+            if (event.target.value === "") {
+              setIntelligenceOverride("0");
```

Comment:
This feels like a classic case of having state in two places. You should just push the useState up into PortalModal, and pass it down via props.